presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreHttpRequestDetails.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore.thrift;
+
+import com.facebook.airlift.json.ObjectMapperProvider;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.thrift.TException;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class ThriftMetastoreHttpRequestDetails
+{
+    private final String url;
+    private final Map<String, String> headers;
+
+    @JsonCreator
+    public ThriftMetastoreHttpRequestDetails(@JsonProperty("url") String url,
+                                             @JsonProperty("headers") Map<String, String> headers)
+    {
+        this.url = url;
+        this.headers = headers;
+    }
+
+    @JsonCreator
+    public static ThriftMetastoreHttpRequestDetails fromString(String url) throws TException
+    {
+        try {
+            return new ObjectMapperProvider().get().readValue(url, ThriftMetastoreHttpRequestDetails.class);
```

Comment:
How can we deserialize an object from URI?

Reply:
The json string is more of a map with all details related to the http request. It consists of the url itself and a map of headers. Added a sample payload in the comment here - https://github.com/prestodb/presto/pull/15694/files#diff-58acfa8c6349c6332fb62f96971ad111e0be36c976740f117394e227449b4812